### PR TITLE
*: enable the CI test for concurrent ddl

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

File: ddl/ddl.go [hunk 1]
```diff
@@ -1194,14 +1194,36 @@ func (d *ddl) SwitchConcurrentDDL(toConcurrentDDL bool) error {
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	var err error
+	defer func() {
+		if err != nil && !toConcurrentDDL {
```
Comment (Benjamin2037) [inline ddl/ddl.go:1201]: concurrentDDL feature can be closed?

**Review by hawkingrei — APPROVED** [review]

File: ddl/ddl.go [hunk 2]
```diff
@@ -1194,14 +1194,36 @@ func (d *ddl) SwitchConcurrentDDL(toConcurrentDDL bool) error {
...
+			sql := fmt.Sprintf("UPDATE HIGH_PRIORITY %[1]s.%[2]s SET VARIABLE_VALUE = %[4]d WHERE VARIABLE_NAME = '%[3]s'",
+				mysql.SystemDB, mysql.GlobalVariablesTable, variable.TiDBEnableMDL, 1)
+			sess, err := d.sessPool.get()
+			if err != nil {
+				if err != nil {
```
Comment (hawkingrei) [inline ddl/ddl.go:1206]: ddl/ddl.go:1206:12: tautological condition: non-nil != nil (nilness)

**Comment by hawkingrei** [comment]
```
        	Error:      	Expected nil, but got: "invalid memory address or nil pointer dereference"
        	Test:       	TestBatchInsertWithOnDuplicate
        	Messages:   	goroutine 289933 [running]:
        	            	github.com/pingcap/tidb/testkit.GetStack(...)
        	            		testkit/asynctestkit.go:76
        	            	github.com/pingcap/tidb/testkit.(*AsyncTestKit).ConcurrentRun.func1.1()
        	            		testkit/asynctestkit.go:107 +0x7b
        	            	panic({0x489e420, 0x7485a00})
        	            		GOROOT/src/runtime/panic.go:884 +0x212
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1()
        	            		executor/adapter.go:438 +0x417
        	            	panic({0x489e420, 0x7485a00})
        	            		GOROOT/src/runtime/panic.go:884 +0x212
        	            	github.com/pingcap/tidb/infoschema.(*SessionExtendedInfoSchema).TableByID(0xc01b335260, 0x2?)
        	            		infoschema/infoschema.go:662 +0xe3
        	            	github.com/pingcap/tidb/session.addTableNameInTableIDField({0x470d060?, 0xc0254d9f10?}, {0x56e1510, 0xc01b335260})
        	            		session/session.go:1059 +0x116
        	            	github.com/pingcap/tidb/session.(*session).tryReplaceWriteConflictError(0xc0095db400?, {0x5683860, 0xc00daa4ea0})
        	            		session/session.go:1032 +0x11a
        	            	github.com/pingcap/tidb/session.(*session).doCommitWithRetry(0xc0095db400, {0x56abe70, 0xc0286493b0})
        	            		session/session.go:967 +0x335
        	            	github.com/pingcap/tidb/session.(*session).CommitTxn(0xc0095db400, {0x56abe70, 0xc028648810})
        	            		session/session.go:1094 +0x258
        	            	github.com/pingcap/tidb/sessiontxn/internal.CommitBeforeEnterNewTxn({0x56abe70, 0xc028648810}, {0x571c108, 0xc0095db400})
        	            		sessiontxn/internal/txn.go:51 +0x113
        	            	github.com/pingcap/tidb/sessiontxn/isolation.(*baseTxnContextProvider).OnInitialize(0xc00a50e3f0, {0x56abe70?, 0xc028648810?}, 0x0)
        	            		sessiontxn/isolation/base.go:83 +0x26d
        	            	github.com/pingcap/tidb/session.(*txnManager).EnterNewTxn(0xc00a50e3c0, {0x56abe70, 0xc028648810}, 0xc0093d5200)
        	            		session/txnmanager.go:147 +0x78
        	            	github.com/pingcap/tidb/sessiontxn.NewTxn({0x56abe70, 0xc028648810}, {0x571c108?, 0xc0095db400?})
        	            		sessiontxn/interface.go:200 +0x91
        	            	github.com/pingcap/tidb/sessiontxn.NewTxnInStmt({0x56abe70, 0xc028648810}, {0x571c108, 0xc0095db400})
        	            		sessiontxn/interface.go:209 +0x3b
        	            	github.com/pingcap/tidb/executor.(*InsertValues).doBatchInsert(0xc006a44a80, {0x56abe70, 0xc028648810})
        	            		executor/insert_common.go:541 +0x138
        	            	github.com/pingcap/tidb/executor.insertRows({0x56abe70, 0xc028648810}, {0x5690de0, 0xc00a615c20})
        	            		executor/insert_common.go:272 +0x5fa
        	            	github.com/pingcap/tidb/executor.(*InsertExec).Next(0xc00a615c20, {0x56abe70, 0xc028648750}, 0xc0286485d0?)
        	            		executor/insert.go:313 +0xf0
        	            	github.com/pingcap/tidb/executor.Next({0x56abe70, 0xc028648750}, {0x56b0320, 0xc00a615c20}, 0xc00a60fe50)
        	            		executor/executor.go:327 +0x55f
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).next(0xc00a369ef0, {0x56abe70, 0xc028648750}, {0x56b0320, 0xc00a615c20}, 0x0?)
        	            		executor/adapter.go:1118 +0x7e
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor(0xc00a369ef0, {0x56abe70, 0xc028648750}, {0x56b0320?, 0xc00a615c20})
        	            		executor/adapter.go:891 +0x611
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelay(0xc00a369ef0, {0x56abe70, 0xc028648750}, {0x56b0320?, 0xc00a615c20?}, 0x0)
        	            		executor/adapter.go:739 +0x29d
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).Exec(0xc00a369ef0, {0x56abe70, 0xc028648750})
        	            		executor/adapter.go:544 +0xf78
        	            	github.com/pingcap/tidb/session.runStmt({0x56abe70, 0xc01b800ea0}, 0xc0095db400, {0x56bd380, 0xc00a369ef0?})
        	            		session/session.go:2341 +0x6c9
        	            	github.com/pingcap/tidb/session.(*session).ExecuteStmt(0xc0095db400, {0x56abe70, 0xc01b800ea0}, {0x56c4050?, 0xc00a680c80?})
        	            		session/session.go:2205 +0xcf6
        	            	github.com/pingcap/tidb/session.(*session).ExecutePreparedStmt(0xc0095db400, {0x56abe70, 0xc01b800ea0}, 0x1, {0xc00a69f180, 0x7, 0x7})
        	            		session/session.go:2511 +0x1a9
        	            	github.com/pingcap/tidb/testkit.(*AsyncTestKit).Exec(0xc01997dce0, {0x56abe70, 0xc01b800ea0}, {0x4f15cdc, 0x48}, {0xc00aaa2310, 0x7, 0x7})
        	            		testkit/asynctestkit.go:164 +0x1c3
        	            	github.com/pingcap/tidb/executor_test.TestBatchInsertWithOnDuplicate.func2({0x56abe70, 0xc01b800ea0}, 0xc009db6798?, {0xc015d0e210, 0x1, 0xc036b10780?})
        	            		executor/write_concurrent_test.go:54 +0xbf
        	            	github.com/pingcap/tidb/testkit.(*AsyncTestKit).ConcurrentRun.func1()
        	            		testkit/asynctestkit.go:112 +0x12a
        	            	created by github.com/pingcap/tidb/testkit.(*AsyncTestKit).ConcurrentRun
        	            		testkit/asynctestkit.go:104 +0xc7
```

**Pull request merged**